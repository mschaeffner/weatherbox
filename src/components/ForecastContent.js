import React from 'react';
import moment from 'moment';


const kelvinToCelsius = (kelvin) => kelvin - 273

export default ({data}) =>
  <div>
    {data.list.map(entry =>
      <div key={entry.dt}>
        {moment.unix(entry.dt).format("DD.MM.YYYY HH:mm")}
        <img alt='icon' src={'http://openweathermap.org/img/w/' + entry.weather[0].icon + '.png'} />
        {entry.weather[0].description + ' and ' + Math.round(kelvinToCelsius(entry.main.temp)) + 'Â°C'}
      </div>
    )}
  </div>
