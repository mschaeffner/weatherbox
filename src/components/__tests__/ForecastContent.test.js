import React from 'react';
import renderer from 'react-test-renderer';
import ForecastContent from '../ForecastContent';


describe('ForecastContent', () => {
  it('should show details for a given forecast data object', () => {
    const component = renderer.create(<ForecastContent data={data} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

const data = {"city":{"id":2950159,"name":"Berlin","coord":{"lat":52.5244,"lon":13.4105},"country":"DE"},"list":[{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":295.64,"pressure":1015.05,"humidity":76,"temp_min":294.089,"temp_max":295.64,"sea_level":1020.65,"grnd_level":1015.05},"wind":{"speed":4.22,"deg":79.5007},"rain":{"3h":0.0049999999999999},"clouds":{"all":68},"dt":1500908400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04d"}],"main":{"temp":293.99,"pressure":1014.18,"humidity":75,"temp_min":292.827,"temp_max":293.99,"sea_level":1019.81,"grnd_level":1014.18},"wind":{"speed":4.87,"deg":67.0011},"rain":{"3h":0.0},"clouds":{"all":68},"dt":1500919200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":803,"main":"Clouds","description":"broken clouds","icon":"04n"}],"main":{"temp":291.95,"pressure":1013.71,"humidity":77,"temp_min":291.171,"temp_max":291.95,"sea_level":1019.34,"grnd_level":1013.71},"wind":{"speed":5.72,"deg":68.0009},"rain":{"3h":0.0},"clouds":{"all":76},"dt":1500930000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":290.4,"pressure":1012.82,"humidity":83,"temp_min":290.015,"temp_max":290.4,"sea_level":1018.51,"grnd_level":1012.82},"wind":{"speed":5.21,"deg":58.0023},"rain":{"3h":0.0049999999999999},"clouds":{"all":64},"dt":1500940800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":804,"main":"Clouds","description":"overcast clouds","icon":"04n"}],"main":{"temp":289.233,"pressure":1011.91,"humidity":88,"temp_min":289.233,"temp_max":289.233,"sea_level":1017.68,"grnd_level":1011.91},"wind":{"speed":6.1,"deg":58.5011},"rain":{"3h":0.0},"clouds":{"all":88},"dt":1500951600,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":288.614,"pressure":1011.99,"humidity":86,"temp_min":288.614,"temp_max":288.614,"sea_level":1017.6,"grnd_level":1011.99},"wind":{"speed":6.81,"deg":52.0026},"rain":{"3h":0.13},"clouds":{"all":88},"dt":1500962400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":290.673,"pressure":1011.37,"humidity":85,"temp_min":290.673,"temp_max":290.673,"sea_level":1017.16,"grnd_level":1011.37},"wind":{"speed":6.47,"deg":36.5018},"rain":{"3h":0.029999999999999},"clouds":{"all":92},"dt":1500973200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":291.094,"pressure":1011.24,"humidity":85,"temp_min":291.094,"temp_max":291.094,"sea_level":1016.97,"grnd_level":1011.24},"wind":{"speed":6.26,"deg":26.5001},"rain":{"3h":0.23},"clouds":{"all":92},"dt":1500984000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":290.18,"pressure":1010.96,"humidity":95,"temp_min":290.18,"temp_max":290.18,"sea_level":1016.68,"grnd_level":1010.96},"wind":{"speed":6.18,"deg":13.0005},"rain":{"3h":0.89},"clouds":{"all":100},"dt":1500994800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":289.62,"pressure":1010.93,"humidity":95,"temp_min":289.62,"temp_max":289.62,"sea_level":1016.63,"grnd_level":1010.93},"wind":{"speed":5.61,"deg":350.503},"rain":{"3h":0.46},"clouds":{"all":92},"dt":1501005600,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":289.59,"pressure":1010.72,"humidity":96,"temp_min":289.59,"temp_max":289.59,"sea_level":1016.53,"grnd_level":1010.72},"wind":{"speed":4.45,"deg":353.002},"rain":{"3h":0.45},"clouds":{"all":100},"dt":1501016400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":289.754,"pressure":1010.7,"humidity":97,"temp_min":289.754,"temp_max":289.754,"sea_level":1016.37,"grnd_level":1010.7},"wind":{"speed":4.91,"deg":348.503},"rain":{"3h":0.24},"clouds":{"all":92},"dt":1501027200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":289.359,"pressure":1010.31,"humidity":97,"temp_min":289.359,"temp_max":289.359,"sea_level":1016.04,"grnd_level":1010.31},"wind":{"speed":5.71,"deg":335.001},"rain":{"3h":0.57},"clouds":{"all":100},"dt":1501038000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"main":{"temp":288.959,"pressure":1010.12,"humidity":99,"temp_min":288.959,"temp_max":288.959,"sea_level":1015.86,"grnd_level":1010.12},"wind":{"speed":7.52,"deg":325.002},"rain":{"3h":5.6},"clouds":{"all":92},"dt":1501048800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":501,"main":"Rain","description":"moderate rain","icon":"10d"}],"main":{"temp":289.501,"pressure":1010.71,"humidity":99,"temp_min":289.501,"temp_max":289.501,"sea_level":1016.47,"grnd_level":1010.71},"wind":{"speed":8.25,"deg":331.506},"rain":{"3h":6.36},"clouds":{"all":92},"dt":1501059600,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":290.115,"pressure":1011.83,"humidity":97,"temp_min":290.115,"temp_max":290.115,"sea_level":1017.47,"grnd_level":1011.83},"wind":{"speed":8.41,"deg":328.0},"rain":{"3h":0.69},"clouds":{"all":92},"dt":1501070400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":290.397,"pressure":1012.63,"humidity":93,"temp_min":290.397,"temp_max":290.397,"sea_level":1018.35,"grnd_level":1012.63},"wind":{"speed":8.71,"deg":328.003},"rain":{"3h":0.4},"clouds":{"all":88},"dt":1501081200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":289.849,"pressure":1013.55,"humidity":94,"temp_min":289.849,"temp_max":289.849,"sea_level":1019.19,"grnd_level":1013.55},"wind":{"speed":7.17,"deg":326.003},"rain":{"3h":0.84},"clouds":{"all":92},"dt":1501092000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":289.073,"pressure":1014.49,"humidity":93,"temp_min":289.073,"temp_max":289.073,"sea_level":1020.12,"grnd_level":1014.49},"wind":{"speed":6.61,"deg":318.504},"rain":{"3h":0.47},"clouds":{"all":80},"dt":1501102800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":288.569,"pressure":1014.83,"humidity":93,"temp_min":288.569,"temp_max":288.569,"sea_level":1020.48,"grnd_level":1014.83},"wind":{"speed":5.72,"deg":312.502},"rain":{"3h":0.09},"clouds":{"all":64},"dt":1501113600,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"02n"}],"main":{"temp":286.738,"pressure":1014.93,"humidity":92,"temp_min":286.738,"temp_max":286.738,"sea_level":1020.68,"grnd_level":1014.93},"wind":{"speed":4.33,"deg":303.0},"rain":{"3h":0.0},"clouds":{"all":8},"dt":1501124400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01d"}],"main":{"temp":288.741,"pressure":1014.88,"humidity":94,"temp_min":288.741,"temp_max":288.741,"sea_level":1020.52,"grnd_level":1014.88},"wind":{"speed":2.86,"deg":288.502},"rain":{"3h":0.0},"clouds":{"all":0},"dt":1501135200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":800,"main":"Clear","description":"clear sky","icon":"01d"}],"main":{"temp":292.994,"pressure":1014.09,"humidity":95,"temp_min":292.994,"temp_max":292.994,"sea_level":1019.67,"grnd_level":1014.09},"wind":{"speed":3.67,"deg":266.003},"rain":{"3h":0.0},"clouds":{"all":0},"dt":1501146000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"main":{"temp":295.652,"pressure":1012.89,"humidity":91,"temp_min":295.652,"temp_max":295.652,"sea_level":1018.55,"grnd_level":1012.89},"wind":{"speed":3.13,"deg":245.006},"rain":{"3h":0.0},"clouds":{"all":12},"dt":1501156800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":293.679,"pressure":1012.27,"humidity":86,"temp_min":293.679,"temp_max":293.679,"sea_level":1017.84,"grnd_level":1012.27},"wind":{"speed":4.92,"deg":242.502},"rain":{"3h":0.3875},"clouds":{"all":92},"dt":1501167600,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":291.696,"pressure":1012.28,"humidity":94,"temp_min":291.696,"temp_max":291.696,"sea_level":1017.91,"grnd_level":1012.28},"wind":{"speed":3.91,"deg":249.001},"rain":{"3h":0.575},"clouds":{"all":92},"dt":1501178400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":290.912,"pressure":1012.27,"humidity":97,"temp_min":290.912,"temp_max":290.912,"sea_level":1017.89,"grnd_level":1012.27},"wind":{"speed":4.26,"deg":234.501},"rain":{"3h":0.4375},"clouds":{"all":92},"dt":1501189200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":290.78,"pressure":1012.56,"humidity":97,"temp_min":290.78,"temp_max":290.78,"sea_level":1018.27,"grnd_level":1012.56},"wind":{"speed":4.92,"deg":257.5},"rain":{"3h":1.0125},"clouds":{"all":92},"dt":1501200000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10n"}],"main":{"temp":290.065,"pressure":1013.56,"humidity":96,"temp_min":290.065,"temp_max":290.065,"sea_level":1019.24,"grnd_level":1013.56},"wind":{"speed":4.77,"deg":289.506},"rain":{"3h":0.024999999999999},"clouds":{"all":88},"dt":1501210800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":288.73,"pressure":1015.6,"humidity":95,"temp_min":288.73,"temp_max":288.73,"sea_level":1021.32,"grnd_level":1015.6},"wind":{"speed":6.01,"deg":286.512},"rain":{"3h":0.0625},"clouds":{"all":92},"dt":1501221600,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":290.338,"pressure":1017.71,"humidity":91,"temp_min":290.338,"temp_max":290.338,"sea_level":1023.41,"grnd_level":1017.71},"wind":{"speed":6.42,"deg":286.5},"rain":{"3h":0.050000000000001},"clouds":{"all":36},"dt":1501232400,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":500,"main":"Rain","description":"light rain","icon":"10d"}],"main":{"temp":292.532,"pressure":1019.21,"humidity":84,"temp_min":292.532,"temp_max":292.532,"sea_level":1024.94,"grnd_level":1019.21},"wind":{"speed":6.71,"deg":285.0},"rain":{"3h":0.012499999999999},"clouds":{"all":12},"dt":1501243200,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"main":{"temp":292.441,"pressure":1020.62,"humidity":73,"temp_min":292.441,"temp_max":292.441,"sea_level":1026.31,"grnd_level":1020.62},"wind":{"speed":6.72,"deg":285.0},"rain":{"3h":0.0},"clouds":{"all":36},"dt":1501254000,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":802,"main":"Clouds","description":"scattered clouds","icon":"03d"}],"main":{"temp":291.377,"pressure":1022.09,"humidity":69,"temp_min":291.377,"temp_max":291.377,"sea_level":1027.82,"grnd_level":1022.09},"wind":{"speed":5.81,"deg":283.001},"rain":{"3h":0.0},"clouds":{"all":32},"dt":1501264800,"sys":{"sunrise":0,"sunset":0},"id":0},{"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02n"}],"main":{"temp":288.424,"pressure":1023.49,"humidity":74,"temp_min":288.424,"temp_max":288.424,"sea_level":1029.29,"grnd_level":1023.49},"wind":{"speed":3.56,"deg":274.502},"rain":{"3h":0.0},"clouds":{"all":12},"dt":1501275600,"sys":{"sunrise":0,"sunset":0},"id":0}]}
