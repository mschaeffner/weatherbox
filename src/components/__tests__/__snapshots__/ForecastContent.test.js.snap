// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastContent should show details for a given forecast data object 1`] = `
<div>
  <div>
    25.07.2017 03:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 23°C
  </div>
  <div>
    25.07.2017 06:00
    <img
      src="http://openweathermap.org/img/w/04d.png"
    />
    broken clouds and 21°C
  </div>
  <div>
    25.07.2017 09:00
    <img
      src="http://openweathermap.org/img/w/04n.png"
    />
    broken clouds and 19°C
  </div>
  <div>
    25.07.2017 12:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 17°C
  </div>
  <div>
    25.07.2017 15:00
    <img
      src="http://openweathermap.org/img/w/04n.png"
    />
    overcast clouds and 16°C
  </div>
  <div>
    25.07.2017 18:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 16°C
  </div>
  <div>
    25.07.2017 21:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 18°C
  </div>
  <div>
    26.07.2017 00:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 18°C
  </div>
  <div>
    26.07.2017 03:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 17°C
  </div>
  <div>
    26.07.2017 06:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 17°C
  </div>
  <div>
    26.07.2017 09:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 17°C
  </div>
  <div>
    26.07.2017 12:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 17°C
  </div>
  <div>
    26.07.2017 15:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 16°C
  </div>
  <div>
    26.07.2017 18:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    moderate rain and 16°C
  </div>
  <div>
    26.07.2017 21:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    moderate rain and 17°C
  </div>
  <div>
    27.07.2017 00:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 17°C
  </div>
  <div>
    27.07.2017 03:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 17°C
  </div>
  <div>
    27.07.2017 06:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 17°C
  </div>
  <div>
    27.07.2017 09:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 16°C
  </div>
  <div>
    27.07.2017 12:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 16°C
  </div>
  <div>
    27.07.2017 15:00
    <img
      src="http://openweathermap.org/img/w/02n.png"
    />
    clear sky and 14°C
  </div>
  <div>
    27.07.2017 18:00
    <img
      src="http://openweathermap.org/img/w/01d.png"
    />
    clear sky and 16°C
  </div>
  <div>
    27.07.2017 21:00
    <img
      src="http://openweathermap.org/img/w/01d.png"
    />
    clear sky and 20°C
  </div>
  <div>
    28.07.2017 00:00
    <img
      src="http://openweathermap.org/img/w/02d.png"
    />
    few clouds and 23°C
  </div>
  <div>
    28.07.2017 03:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 21°C
  </div>
  <div>
    28.07.2017 06:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 19°C
  </div>
  <div>
    28.07.2017 09:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 18°C
  </div>
  <div>
    28.07.2017 12:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 18°C
  </div>
  <div>
    28.07.2017 15:00
    <img
      src="http://openweathermap.org/img/w/10n.png"
    />
    light rain and 17°C
  </div>
  <div>
    28.07.2017 18:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 16°C
  </div>
  <div>
    28.07.2017 21:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 17°C
  </div>
  <div>
    29.07.2017 00:00
    <img
      src="http://openweathermap.org/img/w/10d.png"
    />
    light rain and 20°C
  </div>
  <div>
    29.07.2017 03:00
    <img
      src="http://openweathermap.org/img/w/03d.png"
    />
    scattered clouds and 19°C
  </div>
  <div>
    29.07.2017 06:00
    <img
      src="http://openweathermap.org/img/w/03d.png"
    />
    scattered clouds and 18°C
  </div>
  <div>
    29.07.2017 09:00
    <img
      src="http://openweathermap.org/img/w/02n.png"
    />
    few clouds and 15°C
  </div>
</div>
`;
